### PR TITLE
chore(ci): add ci workflow for spelunker-api

### PR DESCRIPTION
**Pull request by fallenoak**
This PR introduces a CI workflow for `spelunker-api`. We don't have any tests currently, but it does install dependencies, and runs the `lint` script.

### CODE REVIEW

File: .github/workflows/api-ci.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+name: CI
```
Comment (timkurvers) [inline .github/workflows/api-ci.yml:1]: Suppose we might want to rename this to `api-ci` or something at some point, if the web client gets its own workflow.

**Review by timkurvers — APPROVED** [review]
🙌

File: .github/workflows/api-ci.yml (hunk 1 above)
Comment (fallenoak) [reply .github/workflows/api-ci.yml:1]: Enabling required checks creates a problem with more specific workflow names: the conditional on `paths` in `on.push` and `on.pull_request` means this workflow only runs for PRs that modify `packages/spelunker-api/<sub-path-here>`. But I think we want to enable required checks for PRs to ensure only PRs that pass the checks can be merged.

https://github.community/t/feature-request-conditional-required-checks/16761

By naming the workflows the same, we can cheat, and just require eg. `CI / Node 10` regardless of which CI workflow it is. The name of the YAML file is apparently not part of the check key.

It's janky, but until GitHub supports conditional required checks, or we split Spelunker into dedicated repos, it's probably the easiest option.

Comment (timkurvers) [reply .github/workflows/api-ci.yml:1]: Ah right, sounds good! 👍

**Pull request merged**